### PR TITLE
InstanceMutater should handle different required profiles depending on broker

### PR DESCRIPTION
**Pull request by hmlanigan**
## Description of change

There is a different list of lxd profiles for juju lxd machines and juju lxd containers.  Both use "default", but only a juju lxd machine uses the "juju-<modelname>" profile.

Model cache WatchMachines() should not report containers.

Set the machine's modification status to Idle before making any changes to the profiles.

## QA steps

1. export JUJU_DEV_FEATURE_FLAGS=instance-mutater
2. juju bootstrap localhost
3. juju deploy ./testcharms/charm-repo/quantal/lxd-profile-alt --to lxd -n 2
2. wait for both units to be active
3. juju upgrade-charm lxd-profile-alt  --path ./testcharms/charm-repo/quantal/lxd-profile-alt
4. verify that both are successful and that their profiles have been updated

### CODE REVIEW

File: core/cache/model.go [hunk 1]
```diff
@@ -128,19 +131,26 @@ func (m *Model) Machine(machineId string) (*Machine, error) {
 
 // WatchMachines returns a PredicateStringsWatcher to notify about
 // added and removed machines in the model.  The initial event contains
-// a slice of the current machine ids.
-func (m *Model) WatchMachines() *PredicateStringsWatcher {
+// a slice of the current machine ids.  Containers are excluded.
+func (m *Model) WatchMachines() (*PredicateStringsWatcher, error) {
 	m.mu.Lock()
 
+	// Create a compiled regexp to match machines not containers.
```
Comment (hmlanigan) [inline core/cache/model.go]: Fixing the required lxd profiles based on juju lxd machine vs juju lxd container, brought to light the problem of WatchMachines returning containers too.  Makes QA easier to fix now.

File: worker/instancemutater/mutater.go [hunk 2]
```diff
@@ -5,7 +5,6 @@ package instancemutater
 
 import (
 	"fmt"
-
```
Comment (manadart) [inline worker/instancemutater/mutater.go]: Looks like an accidental removal. Goimports seems to be complaining about it.

Comment (hmlanigan) [reply worker/instancemutater/mutater.go]: ty

**Comment by hmlanigan** [comment]
$$merge$$

**Comment by hmlanigan** [comment]
$$merge$$

**Pull request merged**